integration-cli/docker_utils.go

```diff
@@ -569,17 +569,42 @@ func fakeContext(dockerfile string, files map[string]string) (*FakeContext, erro
 	return ctx, nil
 }
 
-type FakeStorage struct {
+// FakeStorage is a static file server. It might be running locally or remotely
+// on test host.
+type FakeStorage interface {
+	Close() error
+	URL() string
+	CtxDir() string
+}
+
+// fakeStorage returns either a local or remote (at daemon machine) file server
+func fakeStorage(files map[string]string) (FakeStorage, error) {
+	if isLocalDaemon {
```

Comment:
Why aren't we just going to use the container-based http server in both cases to avoid having to maintain two separate implementations that at the end of the day perform exactly the same function in the same way?